kong/pdk/ip.lua

```diff
@@ -0,0 +1,51 @@
+local ip = require "resty.mediador.ip"
+
+
+local function new(self)
+  local _IP = {}
+
+  local ips = self.configuration.trusted_ips or {}
+  local n_ips = #ips
+  local trusted_ips = self.table.new(n_ips, 0)
+  local trust_all_ipv4
+  local trust_all_ipv6
+
+  -- This is because we don't support unix: that the ngx_http_realip module
+  -- supports.  Also as an optimization we will only compile trusted ips if
+  -- Kong is not run with the default 0.0.0.0/0, ::/0 aka trust all ip
+  -- addresses settings.
+  for i = 1, n_ips do
+    local address = ips[i]
+
+    if ip.valid(address) then
+      table.insert(trusted_ips, address)
+
+      if address == "0.0.0.0/0" then
+        trust_all_ipv4 = true
+
+      elseif address == "::/0" then
+        trust_all_ipv6 = true
+      end
+    end
+  end
+
+  if #trusted_ips == 0 then
+    function _IP.is_trusted() return false end
+
+  elseif trust_all_ipv4 and trust_all_ipv6 then
+    function _IP.is_trusted() return true end
```

Comment:
this code is hard to read, and not in line with the rest of the codebase (see `else` clause below), can we update this to:

```lua
_IP.is_trusted = function() return true end
```

Reply:
Sure, though changing that single line to another presentation of the same thing is more or less just syntax juggling. This is just copy-pasted from here:
https://github.com/Kong/kong/blob/master/kong/tools/ip.lua

Reply:
> Sure, though changing that single line to another presentation of the same thing is more or less just syntax juggling.

:+1:, not worth it imho.

Reply:
of course it is syntax juggling, since it is style related. The original file didn't have this style issue though. Will send a pr.